packages/coffeescript/plugin/compile-coffeescript.js

```diff
@@ -69,20 +72,30 @@ export class CoffeeCompiler extends CachingCompiler {
       return null;
     }
 
+    let sourceMap = JSON.parse(output.v3SourceMap);
+
     if (source.indexOf('`') !== -1) {
       // If source contains backticks, pass the coffee output through babel-compiler
-      const doubleRoastedCoffee = this.babelCompiler.processOneFileForTarget(inputFile, output.js);
-      if (doubleRoastedCoffee != null && doubleRoastedCoffee.data != null) {
+      const doubleRoastedCoffee =
+        this.babelCompiler.processOneFileForTarget(inputFile, output.js);
+
+      if (doubleRoastedCoffee != null &&
+          doubleRoastedCoffee.data != null) {
         output.js = doubleRoastedCoffee.data;
+
+        // Combine the original CoffeeScript source map with the one
+        // produced by this.babelCompiler.processOneFileForTarget.
+        const smg = new SourceMapGenerator(
+          new SourceMapConsumer(doubleRoastedCoffee.sourceMap));
+        smg.applySourceMap(new SourceMapConsumer(sourceMap));
+        sourceMap = smg.toJSON();
       }
```

Comment:
Yes, this fixed problems with source maps when the JavaScript within backticks expanded to a different number of lines than the original, so that the total number of lines in the source map was more or less than the total number of lines in the compiled code. Definitely necessary!